core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/datastore/request/SearchResourceRequestBuilder.java

```diff
@@ -76,7 +83,29 @@ public final B filterById(String id) {
 	 * @return this builder instance
 	 */
 	public final B filterByIds(Collection<String> ids) {
-		this.ids = ids;
+		this.ids = ImmutableSet.copyOf(ids);
```

Comment:
How about putting this filter into the `optionsBuilder` immediately (just like we do it for all other enum-based filters)?